### PR TITLE
Expose uuids in catalog

### PR DESCRIPTION
**Pull request by MatthewPiatetsky**
For ECOM-5498, ECOM-5499, ECOM-5501

@edx/ecommerce

### CODE REVIEW

File: course_discovery/apps/course_metadata/search_indexes.py [hunk 1]
```diff
@@ -76,8 +76,10 @@ class CourseIndex(BaseCourseIndex, indexes.Indexable):
 
     course_runs = indexes.MultiValueField()
     expected_learning_items = indexes.MultiValueField()
-
     prerequisites = indexes.MultiValueField(faceted=True)
+    authoring_organization_uuids = indexes.MultiValueField()
```
Comment (clintonb) [inline course_discovery/apps/course_metadata/search_indexes.py]: Given that we don't actually use courses on the marketing site, this is a bit premature. Only course runs and programs matter.

File: course_discovery/apps/course_metadata/tests/test_models.py [hunk 2]
```diff
@@ -34,6 +34,9 @@ class CourseTests(TestCase):
     def setUp(self):
         super(CourseTests, self).setUp()
         self.course = factories.CourseFactory()
+        course_run = factories.CourseRunFactory()
```
Comment (clintonb) [inline course_discovery/apps/course_metadata/tests/test_models.py]: Pass the course to the factory.

File: course_discovery/apps/course_metadata/tests/test_models.py [hunk 3]
```diff
@@ -53,6 +56,13 @@ def test_course_run_update_caught_exception(self):
         with mock.patch.object(Course, 'reindex_course_runs', side_effect=Exception):
             self.course.save()
 
+    def test_staff(self):
+        """ Test that the staff property retrieves course staff correctly from course runs """
+        staff = factories.PersonFactory.create_batch(2)
+        self.course.course_runs.all()[0].staff.add(staff[0])
+        self.course.course_runs.all()[0].staff.add(staff[1])
+        self.assertEqual(self.course.staff, set(staff))
+
```
Comment (schenedx) [inline course_discovery/apps/course_metadata/tests/test_models.py]: please add a new test case for staff coming from different course runs

**Comment by schenedx** [comment]
@MatthewPiatetsky just one comment, rest LGTM 👍

File: course_discovery/apps/course_metadata/tests/test_models.py [hunk 4]
```diff
@@ -163,6 +173,16 @@ def test_program_types(self):
         other_program = factories.ProgramFactory(courses=courses)
         self.assertCountEqual(self.course_run.program_types, [program.type.name, other_program.type.name])
 
+    def test_program_types_unique(self):
```
Comment (clintonb) [inline course_discovery/apps/course_metadata/tests/test_models.py]: Why is this here?

Comment (MatthewPiatetsky) [reply course_discovery/apps/course_metadata/tests/test_models.py]: While I was working on this I noticed that if I had more than one program of the same type, I would have "program_types": ["XSeries", "XSeries"], which seems unhelpful/confusing

Comment (clintonb) [reply course_discovery/apps/course_metadata/tests/test_models.py]: Please move this to a separate commit to avoid confusion and maintain a separation of concerns.

Comment (MatthewPiatetsky) [reply course_discovery/apps/course_metadata/tests/test_models.py]: Does that mean it needs a ticket?

File: course_discovery/apps/course_metadata/search_indexes.py (hunk 1 above)
Comment (MatthewPiatetsky) [reply course_discovery/apps/course_metadata/search_indexes.py]: I see, I get it now, but was confused initially because all the stories refer to courses and programs not course runs and programs

File: course_discovery/apps/course_metadata/tests/test_models.py (hunk 4 above)
Comment (clintonb) [reply course_discovery/apps/course_metadata/tests/test_models.py]: Nope. Just a commit.

**Comment by MatthewPiatetsky** [comment]
@clintonb are your comments addressed?

**Comment by clintonb** [comment]
DRY - clean up the duplicated methods. After that, 👍

**Comment by MatthewPiatetsky** [comment]
Ok I put prepare_authoring_organization_uuids in a base class, the other methods were different, so I left them as is.
